packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -1164,32 +1213,179 @@ export function errorHydratingContainer(parentContainer: Container): void {
+      let element = ownerDocument[type];
+      if (element) {
+        resetPersistentInstance(element);
+        resetProperties(element, type, props, rootContainerInstance);
+        precacheFiberNode(internalInstanceHandle, element);
+        updateFiberProps(element, props);
+      } else {
+        element = fallbackInstance;
+        if (type === HEAD && ownerDocument.body) {
+          insertBefore(documentElement, element, ownerDocument.body);
+        } else {
+          appendChild(documentElement, element);
+        }
+      }
+      const before = getOrMutativelyCreateInsertionEdge(element, type);
+      insertBefore(element, insertionFragment, before);
+      return element;
+    }
+    default: {
+      throw new Error(
+        'commitPersistentPlacement was called with an element type that is not supported. This is a bug in React.',
+      );
+    }
   }
 }
 
-export function getInstanceFromScope(
-  scopeInstance: ReactScopeInstance,
-): null | Object {
-  if (enableScopeAPI) {
-    return getFiberFromScopeInstance(scopeInstance);
+export function getInsertionEdge(parent: Instance): ?InsertionInstance {
+  const tagName = parent.tagName.toLowerCase();
+  if (tagName !== HEAD && tagName !== BODY) {
+    // We only need insertionEdges for Persistent Instances that are expected
+    // to have nodes outside the React tree. This is why we can exclude <html>
+    return null;
   }
-  return null;
+  return getOrMutativelyCreateInsertionEdge(parent, tagName);
 }
 
+function getOrMutativelyCreateInsertionEdge(
+  parent: Instance,
+  tag: string,
+): InsertionInstance {
+  if (tag !== HEAD && tag !== BODY) {
+    throw new Error(
+      'getInsertionEdge was called on something other than <head> or <body>. This is a bug in React.',
+    );
+  }
+  let node = parent.firstChild;
+  let beyondNode = null;
+  while (node != null) {
+    switch (node.nodeType) {
+      case COMMENT_NODE: {
+        const data = ((node: any): Comment).data;
+        if (data === INSERTION_MARKER) {
+          return node;
+        } else if (data === SUSPENSE_END_DATA) {
+          // We were inside a suspense boundary and we're looking for an edge
+          // beyond this
+          beyondNode = false;
+        }
+        break;
+      }
+      default: {
+        if (getInstanceFromNodeDOMTree(node)) {
+          beyondNode = null;
+        } else {
+          beyondNode = beyondNode || node;
+        }
+      }
+    }
+    node = node.nextSibling;
+  }
+  // We cast to instance because we are using insertBefore and appendChild which does not accept a Node.
+  // We could expand the type but I don't want to loosen it unecessarily. This will remain safe as long as the
+  // implementation of these insertion methods is safe for Node's which is likely given they proxy to builtins.
+  const marker: Instance = (parent.ownerDocument.createComment(
+    INSERTION_MARKER,
+  ): any);
```

Comment:
thoughts?

Reply:
no longer an issue, started using the dom native function call since that kind of thing is done elsewhere and they indirection was getting inlined anyway